crates/daphne-server/src/roles/helper.rs

```diff
@@ -1,8 +1,23 @@
-// Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
+// Copyright (c) 2025 Cloudflare, Inc. All rights reserved.
 // SPDX-License-Identifier: BSD-3-Clause
 
 use axum::async_trait;
-use daphne::roles::DapHelper;
+use daphne::{
+    messages::{request::AggregationJobRequestHash, AggregationJobId, TaskId},
+    roles::DapHelper,
+    DapError, DapVersion,
+};
 
 #[async_trait]
-impl DapHelper for crate::App {}
+impl DapHelper for crate::App {
+    async fn assert_agg_job_is_immutable(
+        &self,
+        _id: AggregationJobId,
+        _version: DapVersion,
+        _task_id: &TaskId,
+        _req: &AggregationJobRequestHash,
+    ) -> Result<(), DapError> {
+        // the server implementation can't check for this
+        Ok(())
```

Comment:
Fail loudly. Also, why doesn't `App` use this function?
```suggestion
        unreachable!("the server implementation can't check for this")
```

Reply:
I don't want to fail loudly because this step is mandatory, it's not unreachable. But it didn't exist in the server implementation. 

Which means if we fail loudly here we break the server during the transition to the async worker